test/batch/parsers/xUnitParserSpec.js

```diff
@@ -8,87 +8,94 @@ describe('XUnit Parser', () => {
 
 	beforeEach(()=> {
 		xUnitParser = new XUnitParser({
-			repoUrl: 'https://github.com/user/repo',
-			acceptedTestsLocation: '/tree/master/tests',
-			repo:'user/repo'
+			repo:'user/repo',
+			github:{
+				branch:'master'
+			}
 		});
 	});
 
-	it('should return null if file to parse is empty (no test suite)', () => {
+	it('should return a resolved promise w/ null data if file to parse is empty (no test suite)', (done) => {
```

Comment:
Context should be given by nested `describe` calls with intermediary initialisation steps rather than a long `it`.